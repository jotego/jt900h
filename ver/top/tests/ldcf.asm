    ; Load carry flag
    maxmode on
    relaxed on
    org 0
    ld a,0xbf    ; common header
    jp test

data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
test:
    ; LDCF #4, r
    ld wa,0x115
    ldcf 0,a

    ld b,0
    adc b,0
    cp b,1
    jp ne,bad_end

    ldcf 1,a
    jp c,bad_end

    ldcf 8,wa
    jp nc,bad_end
    or ra3,1

    ; LDCF A,r
    ld bc,0xa55
    ld A,1
    ldcf A,c
    jp c,bad_end

    ld A,11
    ldcf A,bc
    jp nc,bad_end
    or ra3,2

    include finish.inc
    end