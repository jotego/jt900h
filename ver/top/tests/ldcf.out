00000101 - 00000000 - 00000000 - 00000000
00000a55 - 00000000 - 00000000 - 00000000
00000007 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
0000007a - 00000000 - 00000000 - 00000000
SR = f801
